### PR TITLE
feat(configs): configure TypeScript build system

### PR DESCRIPTION
**Pull request by connor-baer**
Addresses #516.

## Purpose

We plan to gradually migrate Circuit UI to TypeScript. The first step is to replace Babel with TypeScript as the compiler. This is a potentially breaking change, so we need to do it in v2. 

## Approach and changes

- Install and configure TypeScript 
- Configure all tooling to work with TypeScript: Eslint, Jest, Storybook
- Add basic types, including one for the Emotion theme

## Definition of done

* [x] Development completed
* [x] Reviewers assigned
* [x] Unit and integration tests
* [x] Meets minimum browser support
* [x] Meets accessibility requirements

### CODE REVIEW

File: src/types/theme.ts [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+  tera = 'tera',
+  afterTera = 'afterTera'
+}
+
+type Typography = {
+  fontSize: string;
+  lineHeight: string;
+};
+
+export interface Theme {
```
Comment (fernandofleury) [inline src/types/theme.ts:34]: You can still break these by groups and use `Merge` if you like.

File: src/util/regex.js [hunk 2]
```diff
@@ -13,9 +13,11 @@
  * limitations under the License.
```
Comment (fernandofleury) [inline src/util/regex.js:13]: These are a bit scary

Comment (connor-baer) [reply src/util/regex.js:13]: Eslint suggested these changes. They look safe to me.

File: src/types/theme.ts (hunk 1 above)
Comment (connor-baer) [reply src/types/theme.ts:34]: Is `Merge` a utility type? I couldn't find any documentation for it.

**Pull request merged**